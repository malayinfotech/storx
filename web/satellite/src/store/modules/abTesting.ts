// Copyright (C) 2022 Storx Labs, Inc.
// See LICENSE for copying information.

import { StoreModule } from '@/types/store';
import { MetaUtils } from '@/utils/meta';
import { ABHitAction, ABTestApi, ABTestValues } from '@/types/abtesting';

export const AB_TESTING_ACTIONS = {
    FETCH: 'fetchAbTestValues',
    RESET: 'resetAbTestValues',
    HIT: 'sendHitEvent',
};

export const AB_TESTING_MUTATIONS = {
    SET: 'SET_AB_VALUES',
    INIT: 'SET_AB_INITIALIZED',
};

export class ABTestingState {
    public abTestValues = new ABTestValues();
    public abTestingEnabled = MetaUtils.getMetaContent('ab-testing-enabled') === 'true';
    public abTestingInitialized = false;
}

interface ABTestingContext {
    state: ABTestingState
    commit: (string, ...unknown) => void
    dispatch: (string, ...unknown) => Promise<any> // eslint-disable-line @typescript-eslint/no-explicit-any
}

const {
    FETCH,
    RESET,
    HIT,
} = AB_TESTING_ACTIONS;

const {
    SET,
    INIT,
} = AB_TESTING_MUTATIONS;

export function makeABTestingModule(api: ABTestApi): StoreModule<ABTestingState, ABTestingContext> {
    return {
        state: new ABTestingState(),
        mutations: {
            [SET](state: ABTestingState, values: ABTestValues): void {
                state.abTestValues = values;
            },
            [INIT](state: ABTestingState, isInitialized = true): void {
                state.abTestingInitialized = isInitialized;
            },
        },
        actions: {
            [FETCH]: async function ({ state, commit }: ABTestingContext): Promise<ABTestValues> {
                if (!state.abTestingEnabled)
                    return state.abTestValues;
                const values = await api.fetchABTestValues();

                await commit(SET, values);
                await commit(INIT);

                return values;
            },
            [RESET]: async function ({ commit }: ABTestingContext) {
                await commit(SET, new ABTestValues());
                await commit(INIT, false);
            },
            [HIT]: async function ({ state, dispatch }: ABTestingContext, action: ABHitAction) {
                if (!state.abTestingEnabled) return;
                if (!state.abTestingInitialized) {
                    await dispatch(FETCH);
                }
                switch (action) {
                case ABHitAction.UPGRADE_ACCOUNT_CLICKED:
                    api.sendHit(ABHitAction.UPGRADE_ACCOUNT_CLICKED).catch(_ => {});
                    break;
                }
            },
        },
    };
}
