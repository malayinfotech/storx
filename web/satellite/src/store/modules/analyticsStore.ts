// Copyright (C) 2022 Storx Labs, Inc.
// See LICENSE for copying information.

import { defineStore } from 'pinia';

import { AnalyticsHttpApi } from '@/api/analytics';

export const useAnalyticsStore = defineStore('analytics', () => {
    const analytics: AnalyticsHttpApi = new AnalyticsHttpApi();

    function eventTriggered(eventName: string): void {
        analytics.eventTriggered(eventName).then(r => {}).catch();
    }

    function linkEventTriggered(eventName: string, link: string): void {
        analytics.linkEventTriggered(eventName, link).then(r => {}).catch();
    }

    function pageVisited(pageName: string): void {
        analytics.pageVisit(pageName).then(r => {}).catch();
    }

    return {
        eventTriggered,
        linkEventTriggered,
        pageVisited,
    };
});
