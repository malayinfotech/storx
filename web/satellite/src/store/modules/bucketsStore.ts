// Copyright (C) 2022 Storx Labs, Inc.
// See LICENSE for copying information.

import { defineStore } from 'pinia';
import { reactive } from 'vue';

import { Bucket, BucketCursor, BucketPage, BucketsApi } from '@/types/buckets';
import { BucketsApiGql } from '@/api/buckets';
import { useProjectsStore } from '@/store/modules/projectsStore';

const BUCKETS_PAGE_LIMIT = 7;
const FIRST_PAGE = 1;

export class BucketsState {
    public allBucketNames: string[] = [];
    public cursor: BucketCursor = { limit: BUCKETS_PAGE_LIMIT, search: '', page: FIRST_PAGE };
    public page: BucketPage = { buckets: new Array<Bucket>(), currentPage: 1, pageCount: 1, offset: 0, limit: BUCKETS_PAGE_LIMIT, search: '', totalCount: 0 };
}

export const useBucketsStore = defineStore('buckets', () => {
    const bucketsState = reactive<BucketsState>({
        allBucketNames: [],
        cursor: { limit: BUCKETS_PAGE_LIMIT, search: '', page: FIRST_PAGE },
        page: { buckets: new Array<Bucket>(), currentPage: 1, pageCount: 1, offset: 0, limit: BUCKETS_PAGE_LIMIT, search: '', totalCount: 0 },
    });

    const api: BucketsApi = new BucketsApiGql();

    function setBucketsSearch(search: string): void {
        bucketsState.cursor.search = search;
    }

    function clearBucketsState(): void {
        bucketsState.allBucketNames = [];
        bucketsState.cursor = new BucketCursor('', BUCKETS_PAGE_LIMIT, FIRST_PAGE);
        bucketsState.page = new BucketPage([], '', BUCKETS_PAGE_LIMIT, 0, 1, 1, 0);
    }

    async function fetchBuckets(page: number): Promise<void> {
        const { projectsStore } = useProjectsStore();
        const projectID = projectsStore.selectedProject.id;
        const before = new Date();
        bucketsState.cursor.page = page;

        bucketsState.page = await api.get(projectID, before, bucketsState.cursor);
    }

    async function fetchAllBucketsNames(): Promise<void> {
        const { projectsStore } = useProjectsStore();
        const projectID = projectsStore.selectedProject.id;

        bucketsState.allBucketNames = await api.getAllBucketNames(projectID);
    }

    return {
        bucketsState,
        setBucketsSearch,
        clearBucketsState,
        fetchBuckets,
        fetchAllBucketsNames,
    };
});
