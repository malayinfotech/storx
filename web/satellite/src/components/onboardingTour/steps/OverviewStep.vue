// Copyright (C) 2020 Storx Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="overview-area">
        <h1 class="overview-area__title" aria-roledescription="title">Welcome to Storx {{ titleLabel }}</h1>
        <p class="overview-area__subtitle">Get started using the web browser, or the command line.</p>
        <div class="overview-area__routes">
            <OverviewContainer
                is-web="true"
                title="Start with web browser"
                info="Start uploading files in the browser and instantly see how your data gets distributed over the Storx network around the world."
                button-label="Continue in web ->"
                :on-click="onUploadInBrowserClick"
            />
            <OverviewContainer
                title="Start with Uplink CLI"
                info="The Uplink CLI is a command-line interface tool which allows you to upload and download files from the network, manage permissions and share files."
                button-label="Continue in cli ->"
                :on-click="onUplinkCLIClick"
            />
        </div>
        <p class="overview-area__skip-button" @click="onSkip">
            Skip and go directly to dashboard
        </p>
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import { AnalyticsHttpApi } from '@/api/analytics';
import { RouteConfig } from '@/router';
import { AnalyticsErrorEventSource, AnalyticsEvent } from '@/utils/constants/analyticsEventNames';
import { MetaUtils } from '@/utils/meta';
import { PartneredSatellite } from '@/types/common';
import { MODALS } from '@/utils/constants/appStatePopUps';
import { APP_STATE_MUTATIONS } from '@/store/mutationConstants';
import { PROJECTS_ACTIONS } from '@/store/modules/projects';

import OverviewContainer from '@/components/onboardingTour/steps/common/OverviewContainer.vue';

// @vue/component
@Component({
    components: {
        OverviewContainer,
    },
})
export default class OverviewStep extends Vue {
    public projectDashboardPath = RouteConfig.ProjectDashboard.path;
    public titleLabel = '';

    private readonly analytics: AnalyticsHttpApi = new AnalyticsHttpApi();

    /**
     * Mounted hook after initial render.
     * Sets correct title label.
     */
    public async mounted(): Promise<void> {
        try {
            await this.$store.dispatch(PROJECTS_ACTIONS.CREATE_DEFAULT_PROJECT);
        } catch (error) {
            this.$notify.error(error.message, AnalyticsErrorEventSource.OVERALL_APP_WRAPPER_ERROR);
            return;
        }

        const partneredSatellites = MetaUtils.getMetaContent('partnered-satellites');
        if (!partneredSatellites) {
            this.titleLabel = 'OSP';
            return;
        }

        const partneredSatellitesJSON = JSON.parse(partneredSatellites);
        const isPartnered = partneredSatellitesJSON.find((el: PartneredSatellite) => {
            return el.name === this.satelliteName;
        });
        if (isPartnered) {
            this.titleLabel = 'DCS';
            return;
        }

        this.titleLabel = 'OSP';
    }

    /**
     * Skips onboarding flow.
     */
    public onSkip(): void {
        this.$router.push(this.projectDashboardPath);
        this.$store.commit(APP_STATE_MUTATIONS.UPDATE_ACTIVE_MODAL, MODALS.createProjectPassphrase);
    }

    /**
     * Holds button click logic.
     * Redirects to next step (creating access grant).
     */
    public onUplinkCLIClick(): void {
        this.$router.push(RouteConfig.OnboardingTour.with(RouteConfig.OnbCLIStep).with(RouteConfig.AGName).path);
        this.analytics.linkEventTriggered(AnalyticsEvent.PATH_SELECTED, 'CLI');
        this.analytics.pageVisit(RouteConfig.OnboardingTour.with(RouteConfig.OnbCLIStep).with(RouteConfig.AGName).path);
    }

    /**
     * Redirects to buckets page.
     */
    public onUploadInBrowserClick(): void {
        this.$store.commit(APP_STATE_MUTATIONS.UPDATE_ACTIVE_MODAL, MODALS.createProjectPassphrase);
    }

    /**
     * Returns satellite name.
     */
    private get satelliteName(): string {
        return this.$store.state.appStateModule.satelliteName;
    }
}
</script>

<style scoped lang="scss">
.overview-area {
    display: flex;
    flex-direction: column;
    align-items: center;
    font-family: 'font_regular', sans-serif;

    &__title {
        font-family: 'font_bold', sans-serif;
        color: #14142b;
        font-size: 32px;
        line-height: 39px;
        margin-bottom: 12.5px;
    }

    &__subtitle {
        font-family: 'font_regular', sans-serif;
        font-weight: 400;
        text-align: center;
        color: #354049;
        font-size: 16px;
        line-height: 21px;
    }

    &__routes {
        margin-top: 35px;
        display: flex;
        align-items: center;
        justify-content: center;
        flex-wrap: wrap;
        column-gap: 38px;
        row-gap: 38px;
    }

    &__skip-button {
        margin-top: 58px;
        color: #b7c1ca;
        cursor: pointer;
        text-decoration: underline;

        &:hover {
            text-decoration: underline;
        }
    }
}

@media screen and (max-width: 760px) {

    .overview-area {
        width: 250px;
        text-align: center;
    }
}
</style>
