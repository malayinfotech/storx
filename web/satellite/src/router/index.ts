// Copyright (C) 2019 Storx Labs, Inc.
// See LICENSE for copying information.

import Vue from 'vue';
import Router from 'vue-router';

import { NavigationLink } from '@/types/navigation';
import { MetaUtils } from '@/utils/meta';
import AllDashboardArea from '@/views/all-dashboard/AllDashboardArea.vue';
import MyProjects from '@/views/all-dashboard/components/MyProjects.vue';

import AccessGrants from '@/components/accessGrants/AccessGrants.vue';
import CreateAccessModal from '@/components/accessGrants/CreateAccessModal.vue';
import CreateAccessGrantFlow from '@/components/accessGrants/newCreateFlow/CreateAccessGrantFlow.vue';
import AccountArea from '@/components/account/AccountArea.vue';
import AccountBilling from '@/components/account/billing/BillingArea.vue';
import BillingOverview from '@/components/account/billing/billingTabs/Overview.vue';
import BillingPaymentMethods from '@/components/account/billing/billingTabs/PaymentMethods.vue';
import BillingHistory2 from '@/components/account/billing/billingTabs/BillingHistory.vue';
import BillingCoupons from '@/components/account/billing/billingTabs/Coupons.vue';
import DetailedHistory from '@/components/account/billing/depositAndBillingHistory/DetailedHistory.vue';
import CreditsHistory from '@/components/account/billing/coupons/CouponArea.vue';
import SettingsArea from '@/components/account/SettingsArea.vue';
import Page404 from '@/components/errors/Page404.vue';
import BucketsView from '@/components/objects/BucketsView.vue';
import ObjectsArea from '@/components/objects/ObjectsArea.vue';
import UploadFile from '@/components/objects/UploadFile.vue';
import OnboardingTourArea from '@/components/onboardingTour/OnboardingTourArea.vue';
import PricingPlanStep from '@/components/onboardingTour/steps/PricingPlanStep.vue';
import OnbCLIStep from '@/components/onboardingTour/steps/CLIStep.vue';
import OverviewStep from '@/components/onboardingTour/steps/OverviewStep.vue';
import CreateProject from '@/components/project/CreateProject.vue';
import EditProjectDetails from '@/components/project/EditProjectDetails.vue';
import ProjectDashboard from '@/components/project/dashboard/ProjectDashboard.vue';
import ProjectsList from '@/components/projectsList/ProjectsList.vue';
import ProjectMembersArea from '@/components/team/ProjectMembersArea.vue';
import CLIInstall from '@/components/onboardingTour/steps/cliFlow/CLIInstall.vue';
import APIKey from '@/components/onboardingTour/steps/cliFlow/APIKey.vue';
import CLISetup from '@/components/onboardingTour/steps/cliFlow/CLISetup.vue';
import CreateBucket from '@/components/onboardingTour/steps/cliFlow/CreateBucket.vue';
import UploadObject from '@/components/onboardingTour/steps/cliFlow/UploadObject.vue';
import ListObject from '@/components/onboardingTour/steps/cliFlow/ListObject.vue';
import DownloadObject from '@/components/onboardingTour/steps/cliFlow/DownloadObject.vue';
import ShareObject from '@/components/onboardingTour/steps/cliFlow/ShareObject.vue';
import RegistrationSuccess from '@/components/common/RegistrationSuccess.vue';
import SuccessScreen from '@/components/onboardingTour/steps/cliFlow/SuccessScreen.vue';
import AGName from '@/components/onboardingTour/steps/cliFlow/AGName.vue';
import AGPermissions from '@/components/onboardingTour/steps/cliFlow/AGPermissions.vue';
import BucketDetails from '@/components/objects/BucketDetails.vue';
import NewSettingsArea from '@/components/account/NewSettingsArea.vue';

const ActivateAccount = () => import('@/views/ActivateAccount.vue');
const AuthorizeArea = () => import('@/views/AuthorizeArea.vue');
const DashboardArea = () => import('@/views/DashboardArea.vue');
const ForgotPassword = () => import('@/views/ForgotPassword.vue');
const LoginArea = () => import('@/views/LoginArea.vue');
const RegisterArea = () => import('@/views/registration/RegisterArea.vue');
const ResetPassword = () => import('@/views/ResetPassword.vue');

Vue.use(Router);

/**
 * RouteConfig contains information about all routes and subroutes
 */
export abstract class RouteConfig {
    // root paths
    public static Root = new NavigationLink('/', 'Root');
    public static AllProjectsDashboard = new NavigationLink('/all-projects', 'All Projects');
    public static Login = new NavigationLink('/login', 'Login');
    public static Register = new NavigationLink('/signup', 'Register');
    public static RegisterSuccess = new NavigationLink('/signup-success', 'RegisterSuccess');
    public static Activate = new NavigationLink('/activate', 'Activate');
    public static ForgotPassword = new NavigationLink('/forgot-password', 'Forgot Password');
    public static ResetPassword = new NavigationLink('/password-recovery', 'Reset Password');
    public static Authorize = new NavigationLink('/oauth/v2/authorize', 'Authorize');
    public static Account = new NavigationLink('/account', 'Account');
    public static AccountSettings = new NavigationLink('/account-settings', 'Account Settings');
    public static ProjectDashboard = new NavigationLink('/project-dashboard', 'Dashboard');
    public static Users = new NavigationLink('/project-members', 'Users');
    public static OnboardingTour = new NavigationLink('/onboarding-tour', 'Onboarding Tour');
    public static CreateProject = new NavigationLink('/create-project', 'Create Project');
    public static EditProjectDetails = new NavigationLink('/edit-project-details', 'Edit Project Details');
    public static AccessGrants = new NavigationLink('/access-grants', 'Access');
    public static ProjectsList = new NavigationLink('/projects', 'Projects');
    public static Buckets = new NavigationLink('/buckets', 'Buckets');

    // account child paths
    public static Settings = new NavigationLink('settings', 'Settings');
    public static Settings2 = new NavigationLink('settings', 'Settings 2');
    public static Billing = new NavigationLink('billing', 'Billing');
    public static Billing2 = new NavigationLink('billing', 'Account Billing');
    public static BillingHistory = new NavigationLink('billing-history', 'Billing History');
    public static BillingHistory4 = new NavigationLink('billing-history', 'Billing History 4');
    public static BillingOverview = new NavigationLink('overview', 'Overview');
    // this duplicates the path of BillingOverview so that they can be used interchangeably in BillingArea.vue
    public static BillingOverview2 = new NavigationLink('overview', 'Billing Overview');
    public static BillingPaymentMethods = new NavigationLink('payment-methods', 'Payment Methods');
    // this duplicates the path of BillingPaymentMethods so that they can be used interchangeably in BillingArea.vue
    public static BillingPaymentMethods2 = new NavigationLink('payment-methods', 'Payment Methods 2');
    public static BillingHistory2 = new NavigationLink('billing-history2', 'Billing History 2');
    // this duplicates the path of BillingHistory2 so that they can be used interchangeably in BillingArea.vue
    public static BillingHistory3 = new NavigationLink('billing-history2', 'Billing History 3');
    public static BillingCoupons = new NavigationLink('coupons', 'Coupons');
    public static BillingCoupons2 = new NavigationLink('coupons', 'Billing Coupons');
    public static DepositHistory = new NavigationLink('deposit-history', 'Deposit History');
    public static DepositHistory2 = new NavigationLink('deposit-history', 'Deposit History 2');
    public static CreditsHistory = new NavigationLink('credits-history', 'Credits History');
    public static CreditsHistory2 = new NavigationLink('credits-history', 'Credits History 2');

    // access grant child paths
    public static CreateAccessModal = new NavigationLink('create-access-modal', 'Create Access Modal');
    public static NewCreateAccessModal = new NavigationLink('new-create-access-modal', 'New Create Access Modal');

    // onboarding tour child paths
    public static PricingPlanStep = new NavigationLink('pricing', 'Pricing Plan');
    public static OverviewStep = new NavigationLink('overview', 'Onboarding Overview');
    public static OnbCLIStep = new NavigationLink('cli', 'Onboarding CLI');
    public static AGName = new NavigationLink('ag-name', 'Onboarding AG Name');
    public static AGPermissions = new NavigationLink('ag-permissions', 'Onboarding AG Permissions');
    public static APIKey = new NavigationLink('api-key', 'Onboarding API Key');
    public static CLIInstall = new NavigationLink('cli-install', 'Onboarding CLI Install');
    public static CLISetup = new NavigationLink('cli-setup', 'Onboarding CLI Setup');
    public static CreateBucket = new NavigationLink('create-bucket', 'Onboarding Create Bucket');
    public static UploadObject = new NavigationLink('upload-object', 'Onboarding Upload Object');
    public static ListObject = new NavigationLink('list-object', 'Onboarding List Object');
    public static DownloadObject = new NavigationLink('download-object', 'Onboarding Download Object');
    public static ShareObject = new NavigationLink('share-object', 'Onboarding Share Object');
    public static SuccessScreen = new NavigationLink('success', 'Onboarding Success Screen');

    public static FirstOnboardingStep = this.OverviewStep;

    // objects child paths.
    public static BucketsManagement = new NavigationLink('management', 'Buckets Management');
    public static BucketsDetails = new NavigationLink('details', 'Bucket Details');
    public static UploadFile = new NavigationLink('upload/', 'Objects Upload');
    public static UploadFileChildren = new NavigationLink('*', 'Objects Upload Children');
}

if (MetaUtils.getMetaContent('pricing-packages-enabled') === 'true') {
    RouteConfig.FirstOnboardingStep = RouteConfig.PricingPlanStep;
}

export const notProjectRelatedRoutes = [
    RouteConfig.Login.name,
    RouteConfig.Register.name,
    RouteConfig.RegisterSuccess.name,
    RouteConfig.Activate.name,
    RouteConfig.ForgotPassword.name,
    RouteConfig.ResetPassword.name,
    RouteConfig.Authorize.name,
    RouteConfig.Billing.name,
    RouteConfig.BillingHistory.name,
    RouteConfig.BillingOverview.name,
    RouteConfig.BillingPaymentMethods.name,
    RouteConfig.BillingHistory2.name,
    RouteConfig.BillingCoupons.name,
    RouteConfig.DepositHistory.name,
    RouteConfig.CreditsHistory.name,
    RouteConfig.Settings.name,
];

export const router = new Router({
    mode: 'history',
    routes: [
        {
            path: RouteConfig.Login.path,
            name: RouteConfig.Login.name,
            component: LoginArea,
        },
        {
            path: RouteConfig.Register.path,
            name: RouteConfig.Register.name,
            component: RegisterArea,
        },
        {
            path: RouteConfig.RegisterSuccess.path,
            name: RouteConfig.RegisterSuccess.name,
            component: RegistrationSuccess,
        },
        {
            path: RouteConfig.Activate.path,
            name: RouteConfig.Activate.name,
            component: ActivateAccount,
        },
        {
            path: RouteConfig.ForgotPassword.path,
            name: RouteConfig.ForgotPassword.name,
            component: ForgotPassword,
        },
        {
            path: RouteConfig.ResetPassword.path,
            name: RouteConfig.ResetPassword.name,
            component: ResetPassword,
        },
        {
            path: RouteConfig.Authorize.path,
            name: RouteConfig.Authorize.name,
            component: AuthorizeArea,
        },
        {
            path: RouteConfig.Root.path,
            meta: {
                requiresAuth: true,
            },
            component: DashboardArea,
            children: [
                {
                    path: RouteConfig.Root.path,
                    name: 'default',
                    component: ProjectDashboard,
                },
                {
                    path: RouteConfig.Account.path,
                    name: RouteConfig.Account.name,
                    component: AccountArea,
                    children: [
                        {
                            path: RouteConfig.Settings.path,
                            name: RouteConfig.Settings.name,
                            component: SettingsArea,
                        },
                        {
                            path: RouteConfig.Billing.path,
                            name: RouteConfig.Billing.name,
                            component: AccountBilling,
                            children: [
                                {
                                    path: RouteConfig.BillingOverview.path,
                                    name: RouteConfig.BillingOverview.name,
                                    component: BillingOverview,
                                },
                                {
                                    path: RouteConfig.BillingPaymentMethods.path,
                                    name: RouteConfig.BillingPaymentMethods.name,
                                    component: BillingPaymentMethods,
                                },
                                {
                                    path: RouteConfig.BillingHistory2.path,
                                    name: RouteConfig.BillingHistory2.name,
                                    component: BillingHistory2,
                                },
                                {
                                    path: RouteConfig.BillingCoupons.path,
                                    name: RouteConfig.BillingCoupons.name,
                                    component: BillingCoupons,
                                },
                            ],
                        },
                        {
                            path: RouteConfig.BillingHistory.path,
                            name: RouteConfig.BillingHistory.name,
                            component: DetailedHistory,
                        },
                        {
                            path: RouteConfig.DepositHistory.path,
                            name: RouteConfig.DepositHistory.name,
                            component: DetailedHistory,
                        },
                        {
                            path: RouteConfig.CreditsHistory.path,
                            name: RouteConfig.CreditsHistory.name,
                            component: CreditsHistory,
                        },
                    ],
                },
                {
                    path: RouteConfig.ProjectDashboard.path,
                    name: RouteConfig.ProjectDashboard.name,
                    component: ProjectDashboard,
                },
                {
                    path: RouteConfig.Users.path,
                    name: RouteConfig.Users.name,
                    component: ProjectMembersArea,
                },
                {
                    path: RouteConfig.OnboardingTour.path,
                    name: RouteConfig.OnboardingTour.name,
                    component: OnboardingTourArea,
                    children: [
                        {
                            path: RouteConfig.PricingPlanStep.path,
                            name: RouteConfig.PricingPlanStep.name,
                            component: PricingPlanStep,
                        },
                        {
                            path: RouteConfig.OverviewStep.path,
                            name: RouteConfig.OverviewStep.name,
                            component: OverviewStep,
                        },
                        {
                            path: RouteConfig.OnbCLIStep.path,
                            name: RouteConfig.OnbCLIStep.name,
                            component: OnbCLIStep,
                            children: [
                                {
                                    path: RouteConfig.AGName.path,
                                    name: RouteConfig.AGName.name,
                                    component: AGName,
                                },
                                {
                                    path: RouteConfig.AGPermissions.path,
                                    name: RouteConfig.AGPermissions.name,
                                    component: AGPermissions,
                                },
                                {
                                    path: RouteConfig.APIKey.path,
                                    name: RouteConfig.APIKey.name,
                                    component: APIKey,
                                },
                                {
                                    path: RouteConfig.CLIInstall.path,
                                    name: RouteConfig.CLIInstall.name,
                                    component: CLIInstall,
                                },
                                {
                                    path: RouteConfig.CLISetup.path,
                                    name: RouteConfig.CLISetup.name,
                                    component: CLISetup,
                                },
                                {
                                    path: RouteConfig.CreateBucket.path,
                                    name: RouteConfig.CreateBucket.name,
                                    component: CreateBucket,
                                },
                                {
                                    path: RouteConfig.UploadObject.path,
                                    name: RouteConfig.UploadObject.name,
                                    component: UploadObject,
                                },
                                {
                                    path: RouteConfig.ListObject.path,
                                    name: RouteConfig.ListObject.name,
                                    component: ListObject,
                                },
                                {
                                    path: RouteConfig.DownloadObject.path,
                                    name: RouteConfig.DownloadObject.name,
                                    component: DownloadObject,
                                },
                                {
                                    path: RouteConfig.ShareObject.path,
                                    name: RouteConfig.ShareObject.name,
                                    component: ShareObject,
                                },
                                {
                                    path: RouteConfig.SuccessScreen.path,
                                    name: RouteConfig.SuccessScreen.name,
                                    component: SuccessScreen,
                                },
                            ],
                        },
                    ],
                },
                {
                    path: RouteConfig.CreateProject.path,
                    name: RouteConfig.CreateProject.name,
                    component: CreateProject,
                },
                {
                    path: RouteConfig.EditProjectDetails.path,
                    name: RouteConfig.EditProjectDetails.name,
                    component: EditProjectDetails,
                },
                {
                    path: RouteConfig.AccessGrants.path,
                    name: RouteConfig.AccessGrants.name,
                    component: AccessGrants,
                    children: [
                        {
                            path: RouteConfig.CreateAccessModal.path,
                            name: RouteConfig.CreateAccessModal.name,
                            component: CreateAccessModal,
                        },
                        {
                            path: RouteConfig.NewCreateAccessModal.path,
                            name: RouteConfig.NewCreateAccessModal.name,
                            component: CreateAccessGrantFlow,
                        },
                    ],
                },
                {
                    path: RouteConfig.ProjectsList.path,
                    name: RouteConfig.ProjectsList.name,
                    component: ProjectsList,
                },
                {
                    path: RouteConfig.Buckets.path,
                    name: RouteConfig.Buckets.name,
                    component: ObjectsArea,
                    children: [
                        {
                            path: RouteConfig.BucketsManagement.path,
                            name: RouteConfig.BucketsManagement.name,
                            component: BucketsView,
                            props: true,
                        },
                        {
                            path: RouteConfig.BucketsDetails.path,
                            name: RouteConfig.BucketsDetails.name,
                            component: BucketDetails,
                            props: true,
                        },
                        {
                            path: RouteConfig.UploadFile.path,
                            name: RouteConfig.UploadFile.name,
                            component: UploadFile,
                            children: [
                                {
                                    path: RouteConfig.UploadFileChildren.path,
                                    name: RouteConfig.UploadFileChildren.name,
                                    component: UploadFile,
                                },
                            ],
                        },
                    ],
                },
            ],
        },
        {
            path: RouteConfig.AllProjectsDashboard.path,
            meta: {
                requiresAuth: true,
            },
            component: AllDashboardArea,
            children: [
                {
                    path: RouteConfig.Root.path,
                    name: RouteConfig.AllProjectsDashboard.name,
                    component: MyProjects,
                },
                {
                    path: RouteConfig.AccountSettings.path,
                    name: RouteConfig.AccountSettings.name,
                    component: AccountArea,
                    children: [
                        {
                            path: RouteConfig.Settings2.path,
                            name: RouteConfig.Settings2.name,
                            component: NewSettingsArea,
                        },
                        {
                            path: RouteConfig.Billing2.path,
                            name: RouteConfig.Billing2.name,
                            component: AccountBilling,
                            children: [
                                {
                                    path: RouteConfig.BillingOverview2.path,
                                    name: RouteConfig.BillingOverview2.path,
                                    component: BillingOverview,
                                },
                                {
                                    path: RouteConfig.BillingPaymentMethods2.path,
                                    name: RouteConfig.BillingPaymentMethods2.name,
                                    component: BillingPaymentMethods,
                                },
                                {
                                    path: RouteConfig.BillingHistory3.path,
                                    name: RouteConfig.BillingHistory3.name,
                                    component: BillingHistory2,
                                },
                                {
                                    path: RouteConfig.BillingCoupons2.path,
                                    name: RouteConfig.BillingCoupons2.name,
                                    component: BillingCoupons,
                                },
                            ],
                        },
                        {
                            path: RouteConfig.BillingHistory4.path,
                            name: RouteConfig.BillingHistory4.name,
                            component: DetailedHistory,
                        },
                        {
                            path: RouteConfig.DepositHistory2.path,
                            name: RouteConfig.DepositHistory2.name,
                            component: DetailedHistory,
                        },
                        {
                            path: RouteConfig.CreditsHistory2.path,
                            name: RouteConfig.CreditsHistory2.name,
                            component: CreditsHistory,
                        },
                    ],
                },
            ],
        },
        {
            path: '*',
            name: '404',
            component: Page404,
        },
    ],
});
