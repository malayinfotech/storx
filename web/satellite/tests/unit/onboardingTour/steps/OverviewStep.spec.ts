// Copyright (C) 2020 Storx Labs, Inc.
// See LICENSE for copying information.

import Vuex from 'vuex';
import { createLocalVue, shallowMount } from '@vue/test-utils';

import { router } from '@/router';
import { appStateModule } from '@/store/modules/appState';

import OverviewStep from '@/components/onboardingTour/steps/OverviewStep.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

const store = new Vuex.Store({ modules: { appStateModule } });

// TODO: figure out how to fix the test
xdescribe('OverviewStep.vue', (): void => {
    it('renders correctly', (): void => {
        const wrapper = shallowMount(OverviewStep, {
            localVue,
            router,
            store,
        });

        expect(wrapper).toMatchSnapshot();
    });
});
