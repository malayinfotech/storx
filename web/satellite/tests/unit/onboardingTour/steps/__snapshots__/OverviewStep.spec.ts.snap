// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewStep.vue renders correctly 1`] = `
<div class="overview-area">
  <h1 aria-roledescription="title" class="overview-area__title">Welcome to Storx </h1>
  <p class="overview-area__subtitle">Get started using the web browser, or the command line.</p>
  <div class="overview-area__routes">
    <overviewcontainer-stub isweb="true" title="Start with web browser" info="Start uploading files in the browser and instantly see how your data gets distributed over the Storx network around the world." buttonlabel="Continue in web ->" onclick="[Function]"></overviewcontainer-stub>
    <overviewcontainer-stub title="Start with Uplink CLI" info="The Uplink CLI is a command-line interface tool which allows you to upload and download files from the network, manage permissions and share files." buttonlabel="Continue in cli ->" onclick="[Function]"></overviewcontainer-stub>
  </div>
  <p class="overview-area__skip-button">
    Skip and go directly to dashboard
  </p>
</div>
`;
